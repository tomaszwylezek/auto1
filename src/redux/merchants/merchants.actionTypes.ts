import { NAME } from 'redux/merchants/merchants.constants';

export const GET_MERCHANTS: string = `${NAME}/GET_MERCHANTS`;
