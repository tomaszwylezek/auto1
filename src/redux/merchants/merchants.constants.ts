export const NAME: string = 'merchants';
