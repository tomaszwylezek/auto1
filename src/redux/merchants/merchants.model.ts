import { IMerchant } from 'model';

export type IMerchantsReducer = IMerchant[];
