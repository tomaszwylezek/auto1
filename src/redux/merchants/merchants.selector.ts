import { createSelector } from 'reselect';
import { IMerchant, IStoreState } from 'model';

export const getMerchants = createSelector<IStoreState, IMerchant[], IMerchant[]>(
  [(state: IStoreState): IMerchant[] => state.merchants],
  (merchants: IMerchant[]): IMerchant[] => merchants
);

export const getMerchant = createSelector<IStoreState, string, IMerchant[], string, IMerchant>(
  getMerchants,
  (_: IStoreState, id: string): string => id,
  (merchants: IMerchant[], id: string): IMerchant =>
    merchants.find(({ id: merchantId }: IMerchant) => id === merchantId)
);
