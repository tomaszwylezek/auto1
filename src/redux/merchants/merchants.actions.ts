import { IAction } from 'model/common';
import { ADD_MERCHANT, DELETE_MERCHANT, EDIT_MERCHANT, GET_MERCHANTS } from 'redux/merchants/merchants.actionTypes';
import { IMerchant } from 'model';

export const getMerchants = (): IAction<IMerchant[]> => ({
  type: GET_MERCHANTS,
  payload: [
    {
      id: '1',
      firstname: 'Tomasz',
      lastname: 'Wylezek',
      avatarUrl: '',
      email: 'xyz@gmail.com',
      phone: '555-555-555',
      hasPremium: true,
      bids: [
        {
          id: '1',
          amount: 12,
          carTitle: 'Subaru',
          created: new Date('Sat Jul 06 2019 10:31:54 GMT+0200 (Central European Summer Time)').toString()
        },
        {
          id: '2',
          amount: 15,
          carTitle: 'Subaru',
          created: new Date(
            +new Date('Sat Jul 06 2019 10:31:54 GMT+0200 (Central European Summer Time)') + 10000
          ).toString()
        }
      ]
    },
    {
      id: '2',
      firstname: 'Grzegorz',
      lastname: 'Wylezek',
      avatarUrl: '',
      email: 'xyz2@gmail.com',
      phone: '665-555-555',
      hasPremium: true,
      bids: [
        {
          id: '3',
          amount: 12,
          carTitle: 'Toyota',
          created: new Date('Sat Jul 06 2019 10:31:54 GMT+0200 (Central European Summer Time)').toString()
        }
      ]
    },
    {
      id: '3',
      firstname: 'Michal',
      lastname: 'Wylezek',
      avatarUrl: '',
      email: 'xyz2@gmail.com',
      phone: '665-555-555',
      hasPremium: true,
      bids: [
        {
          id: '3',
          amount: 12,
          carTitle: 'Toyota',
          created: new Date('Sat Jul 06 2019 10:31:54 GMT+0200 (Central European Summer Time)').toString()
        }
      ]
    },
    {
      id: '4',
      firstname: 'Adam',
      lastname: 'Wylezek',
      avatarUrl: '',
      email: 'xyz2@gmail.com',
      phone: '665-555-555',
      hasPremium: true,
      bids: [
        {
          id: '3',
          amount: 12,
          carTitle: 'Toyota',
          created: new Date('Sat Jul 06 2019 10:31:54 GMT+0200 (Central European Summer Time)').toString()
        }
      ]
    },
    {
      id: '5',
      firstname: 'Lajla',
      lastname: 'Wylezek',
      avatarUrl: '',
      email: 'xyz2@gmail.com',
      phone: '665-555-555',
      hasPremium: true,
      bids: [
        {
          id: '3',
          amount: 12,
          carTitle: 'Toyota',
          created: new Date('Sat Jul 06 2019 10:31:54 GMT+0200 (Central European Summer Time)').toString()
        }
      ]
    },
    {
      id: '6',
      firstname: 'Lidia',
      lastname: 'Wylezek',
      avatarUrl: '',
      email: 'xyz2@gmail.com',
      phone: '665-555-555',
      hasPremium: true,
      bids: [
        {
          id: '3',
          amount: 12,
          carTitle: 'Toyota',
          created: new Date('Sat Jul 06 2019 10:31:54 GMT+0200 (Central European Summer Time)').toString()
        }
      ]
    },
    {
      id: '7',
      firstname: 'Leokadia',
      lastname: 'Wylezek',
      avatarUrl: '',
      email: 'xyz2@gmail.com',
      phone: '665-555-555',
      hasPremium: true,
      bids: [
        {
          id: '3',
          amount: 12,
          carTitle: 'Toyota',
          created: new Date('Sat Jul 06 2019 10:31:54 GMT+0200 (Central European Summer Time)').toString()
        }
      ]
    },
    {
      id: '8',
      firstname: 'Iwona',
      lastname: 'Wylezek',
      avatarUrl: '',
      email: 'xyz2@gmail.com',
      phone: '665-555-555',
      hasPremium: true,
      bids: [
        {
          id: '3',
          amount: 12,
          carTitle: 'Toyota',
          created: new Date('Sat Jul 06 2019 10:31:54 GMT+0200 (Central European Summer Time)').toString()
        }
      ]
    },
    {
      id: '9',
      firstname: 'Grzegorz',
      lastname: 'Wylezek',
      avatarUrl: '',
      email: 'xyz2@gmail.com',
      phone: '665-555-555',
      hasPremium: true,
      bids: [
        {
          id: '3',
          amount: 12,
          carTitle: 'Toyota',
          created: new Date('Sat Jul 06 2019 10:31:54 GMT+0200 (Central European Summer Time)').toString()
        }
      ]
    },
    {
      id: '4322',
      firstname: 'Ania',
      lastname: 'Wylezek',
      avatarUrl: '',
      email: 'xyz2@gmail.com',
      phone: '665-555-555',
      hasPremium: true,
      bids: [
        {
          id: '3',
          amount: 12,
          carTitle: 'Toyota',
          created: new Date('Sat Jul 06 2019 10:31:54 GMT+0200 (Central European Summer Time)').toString()
        }
      ]
    },
    {
      id: '2432432',
      firstname: 'Seba',
      lastname: 'Wylezek',
      avatarUrl: '',
      email: 'xyz2@gmail.com',
      phone: '665-555-555',
      hasPremium: true,
      bids: [
        {
          id: '3',
          amount: 12,
          carTitle: 'Toyota',
          created: new Date('Sat Jul 06 2019 10:31:54 GMT+0200 (Central European Summer Time)').toString()
        }
      ]
    }
  ]
});

export const addMerchant = (merchant: IMerchant): IAction<IMerchant> => ({
  type: ADD_MERCHANT,
  payload: merchant
});

export const editMerchant = (merchant: IMerchant): IAction<IMerchant> => ({
  type: EDIT_MERCHANT,
  payload: merchant
});

export const deleteMerchant = (id: string): IAction<string> => ({
  type: DELETE_MERCHANT,
  payload: id
});
