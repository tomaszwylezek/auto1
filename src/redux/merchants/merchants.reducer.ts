import { handleActions } from 'redux-actions';
import { Reducer } from 'redux';
import { IMerchantsReducer } from 'redux/merchants/merchants.model';
import { GET_MERCHANTS } from 'redux/merchants/merchants.actionTypes';
import { IMerchant } from 'model';

const defaultState: IMerchantsReducer = [];

export const merchantsReducer: Reducer<IMerchantsReducer> = handleActions(
  {
    [GET_MERCHANTS]: (state: IMerchantsReducer, { payload }: { payload: IMerchant[] }): IMerchantsReducer => payload
  },
  defaultState
);
