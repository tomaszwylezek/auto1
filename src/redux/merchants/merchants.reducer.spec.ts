import { merchantsReducer } from 'redux/merchants/merchants.reducer';

describe('Merchants reducer', (): void => {
  it('initialize', (): void => {
    expect(
      merchantsReducer(undefined, {
        type: ''
      })
    ).toMatchSnapshot();
  });
});
