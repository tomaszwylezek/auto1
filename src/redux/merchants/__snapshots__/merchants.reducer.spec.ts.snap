// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Merchants reducer initialize 1`] = `Array []`;
